roqoqo-qasm/src/backend.rs

```diff
@@ -75,7 +83,21 @@ impl Backend {
     ) -> Result<String, RoqoqoBackendError> {
         let mut definitions: String = "".to_string();
         let mut data: String = "".to_string();
-        let mut qasm_string = String::from("OPENQASM 2.0;\n\n");
+        let mut qasm_string = String::from("OPENQASM ");
+        let (qubits, _bits) = if self.qasm_version == "2.0" {
+            qasm_string.push_str("2.0;\n\n");
+            ("qreg", "creg")
+        } else if self.qasm_version == "3.0" {
+            qasm_string.push_str("3.0;\n\n");
+            ("qubit", "bits")
+        } else {
+            return Err(RoqoqoBackendError::GenericError {
+                msg: format!(
+                    "Version for OpenQASM used is neither 2.0 nor 3.0: {}",
+                    self.qasm_version
+                ),
+            });
+        };
```

Comment:
Wouldn't it be better to put out this error during the class variable instantiation? Basically the same check but on line 59. So that the user knows immediately if the backend that they have built is correct or not.

It's just a preference of mine, nothing important really.